extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -473,41 +475,42 @@ export class AmpA4A extends AMP.BaseElement {
 
   /**
    * Try to validate creative is AMP through crypto signature.
-   * @param {!ArrayBuffer} creative
-   * @param {?ArrayBuffer} signature
-   * @return {!Promise<boolean>}
+   * @param {!ArrayBuffer} creative  Bytes of the entire signed creative.
+   * @param {?ArrayBuffer} signature  Bytes for creative signature (decoded from
+   *   base64, if necessary.)
+   * @return {!Promise<ArrayBuffer>}  Promise to a guaranteed-valid AMP creative
+   *   or null if the creative is unsigned or invalid.
    * @private
    */
   validateAdResponse_(creative, signature) {
     // Validate when we have a signature and we have native crypto.
-    if (signature && verifySignatureIsAvailable()) {
-      try {
-        // Among other things, the signature might not be proper base64.
-        // TODO(a4a-cam): This call used to be missing the conversion
-        // from ArrayBuffer to Uint8Array.  Strangely, that didn't cause
-        // any unit tests to fail, either locally or on Travis.  That
-        // indicates that the tests are too weak or aren't reporting
-        // correctly.  Check out and fix the tests.
-        return verifySignature(
-            new Uint8Array(creative), signature, publicKeyInfos);
-      } catch (e) {
-      }
+    if (!signature) {
+      // Guaranteed not a AMP creative.
+      return Promise.resolve(null);
```

Comment:
No.  The problem is that we actually have a number of cases we want to distinguish:
- Valid AMP creative
- Invalid creative / can't validate
- Creative didn't fill
- Error

No signature could still be a valid, but non-AMP creative.  We don't want to report it as an error -- we just want to avoid rendering it in shadow DOM and fall through to let the iframe rendering path handle it.